runtime/modules/json/core/src/test/java/io/atlasmap/json/core/JsonFieldWriterTest.java

```diff
@@ -626,4 +635,565 @@ private String prettyPrintJson(String json) throws IOException {
+        when(session.head().getSourceField()).thenReturn(jsonField);
+
+        Audits audits = new Audits();
+        when(session.getAudits()).thenReturn(audits);
+        reader.read(session);
+        return session;
+    }
+
+    private void testRangeOutValue(Path path, String fieldPath, Object testObject, FieldType inputFieldType, FieldType outputFieldType)
+            throws Exception, IOException, JsonGenerationException, JsonMappingException, AtlasException {
+        write(path, fieldPath, testObject, inputFieldType);
+
+        AtlasInternalSession session = read(path, outputFieldType, fieldPath);
+
+        assertEquals(null, session.head().getSourceField().getValue());
+        assertEquals(1, session.getAudits().getAudit().size());
+        assertEquals("Failed to convert field value '" + testObject.toString() + "' into type '" + outputFieldType.value().toUpperCase() + "'", session.getAudits().getAudit().get(0).getMessage());
+        assertEquals(testObject.toString(), session.getAudits().getAudit().get(0).getValue());
+        assertEquals(AuditStatus.ERROR, session.getAudits().getAudit().get(0).getStatus());
+    }
+
+    @Test
+    public void testJsonFieldDoubleMaxRangeOut() throws Exception {
+        Path path = Paths.get("target" + File.separator + "test-write-field-double-max-range-out.json");
+        String fieldPath = "/primitiveValue";
+        Object testObject = "1.7976931348623157E309";
+        FieldType inputFieldType = FieldType.STRING;
+        FieldType outputFieldType = FieldType.DOUBLE;
+
+        testRangeOutValue(path, fieldPath, testObject, inputFieldType, outputFieldType);
+    }
+
+    @Test
+    public void testJsonFieldDoubleMinRangeOut() throws Exception {
+        Path path = Paths.get("target" + File.separator + "test-write-field-double-min-range-out.json");
+        String fieldPath = "/primitiveValue";
+        Object testObject = "4.9E-325";
+        FieldType inputFieldType = FieldType.STRING;
+        FieldType outputFieldType = FieldType.DOUBLE;
+
+        testMinRangeOutValue(path, fieldPath, testObject, inputFieldType, outputFieldType, 0.0);
+    }
+
+    private void testMinRangeOutValue(Path path, String fieldPath, Object testObject, FieldType inputFieldType, FieldType outputFieldType, Object expectedValue)
+            throws Exception, IOException, JsonGenerationException, JsonMappingException, AtlasException {
+        write(path, fieldPath, testObject, inputFieldType);
+
+        AtlasInternalSession session = read(path, outputFieldType, fieldPath);
+
+        assertEquals(expectedValue, session.head().getSourceField().getValue());
+        assertEquals(0, session.getAudits().getAudit().size());
+    }
+
+    @Test
+    public void testJsonFieldFloatMaxRangeOut() throws Exception {
+        Path path = Paths.get("target" + File.separator + "test-write-field-float-max-range-out.json");
+        String fieldPath = "/primitiveValue";
```

Comment:
Changes look good Joey. Can you put these values directly on the method call, there is not need to declare a variable for these. Similar to what you have done the JSONFieldReader

Reply:
Done. Please review. Thanks